include/fc/io/cfile.hpp

```diff
@@ -0,0 +1,141 @@
+#pragma once
+#include <fc/filesystem.hpp>
+#include <cstdio>
+
+namespace fc {
+
+namespace detail {
+   using unique_file = std::unique_ptr<FILE, decltype( &fclose )>;
+}
+
+class cfile_datastream;
+
+/**
+ * Wrapper for c-file access that provides a similar interface as fstream without all the overhead of std streams.
+ * std::runtime_error exception thrown for errors.
+ */
+class cfile {
+public:
+   cfile()
+     : _file(nullptr, &fclose)
+   {}
+
+   void set_file_path( fc::path file_path ) {
+      _file_path = std::move( file_path );
+   }
+
+   fc::path get_file_path() const {
+      return _file_path;
+   }
+
+   bool is_open() const { return _open; }
+
+   size_t file_size() const {
```

Comment:
stat() of an open file can result in unexpected behavior, up to and including loss of the contents of the file.  Consider requiring the file be closed for file_size to work.

Reply:
We were actually using it in block_log on open files. I'll fix that on the PR to incorporate this into block_log.cpp.